packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -372,16 +356,37 @@ export class TokenRatesController extends StaticIntervalPollingController<
     );
   }
 
+  #subscribeToAccountChange() {
+    this.messagingSystem.subscribe(
+      'AccountsController:selectedEvmAccountChange',
+      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      async (selectedAccount) => {
+        if (this.#selectedAccountId !== selectedAccount.id) {
+          this.#selectedAccountId = selectedAccount.id;
+          if (this.#pollState === PollState.Active) {
+            await this.updateExchangeRates();
+          }
+        }
+      },
+    );
+  }
+
   /**
    * Get the user's tokens for the given chain.
    *
    * @param chainId - The chain ID.
    * @returns The list of tokens addresses for the current chain
    */
   #getTokenAddresses(chainId: Hex): Hex[] {
-    const tokens = this.#allTokens[chainId]?.[this.#selectedAddress] || [];
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getAccount',
+      this.#selectedAccountId,
+    );
+    const tokens =
+      this.#allTokens[chainId]?.[selectedAccount?.address ?? ''] || [];
     const detectedTokens =
-      this.#allDetectedTokens[chainId]?.[this.#selectedAddress] || [];
+      this.#allDetectedTokens[chainId]?.[selectedAccount?.address ?? ''] || [];
```

Comment:
Maybe add a `const selectedAddress = selectedAccount?.address ?? ''` and use it in `this.#allTokens` and `this.#allDetectedTokens`?